pkg/stub/handler.go

```diff
@@ -562,8 +583,14 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 			(srcfg.Spec.SamplesRegistry == "" || srcfg.Spec.SamplesRegistry == "registry.redhat.io") &&
 			!srcfg.ConditionTrue(v1alpha1.ImportCredentialsExist) &&
 			h.registrySecret == nil {
-			err := fmt.Errorf("Cannot create rhel imagestreams to registry.redhat.io without the credentials being available: %#v", srcfg)
-			return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionFalse, err, "%v")
+			if h.waitingForCredential == nil {
+				h.waitingForCredential = srcfg
+				err := fmt.Errorf("Cannot create rhel imagestreams to registry.redhat.io without the credentials being available: %#v", srcfg)
+				return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionFalse, err, "%v")
+			} else {
+				// already recorded need, simply wait for credential to appear
```

Comment:
don't we always want to update h.waitingForCredential?

If I do the following sequence:

1) create the samplesresource (such that it needs a secret)
  - we cache the samplesresource and report an error
2) update the samplesresource (such that it still needs a secret, but perhaps i update the skiplist)
 - currently we ignore that update entirely(?)
3) i create the missing secret
  - we create the samples from the cached value from (1), not the current value from (2).

(I haven't walked the logic carefully so perhaps i'm missing something in the flow where we get the updated samplesresource before doing the samples create)

Reply:
Interesting points  ... taking them on not in sequential order, but by the importance I took for each of them ... high level summary, the answer to your question is yes, but not for the reasons you noted ... at least IMO / AFAIK

- we never create the samples from the cached SamplesResource

neither the traditional `samplesResource` cache, or the new `waitingForCredentials` cache

 ... at most, at most,  cached SamplesResource drives status condition updates of a not entirely ready variety

-  So to build upon your sequence
    1) SR event comes in for rhel, no cred, so cache in `waitingForCredentials`, but event otherwise ignored
    2) SR event updates skipped, still no cred, event still ignored
    3) secret created ... `waitingForCredentials` used to create the ImportCredentialsExists condition
   4) that `ImportCredentialsExists` update leads to a new SR event, which has the updates to the skipped list
    5) now that the cred exists, process the samples
    6) assuming successful, `samplesRegistry` cache item is updated.

So ... why do we still need to update waitingForCredentials then :-) ?

It is the config map for spec validation   ....  suppose they updated the arch's instead of the skipped list before we laid down the initial set of samples ... we want to work off the latest config there.

Reply:
yeah, so I think you're saying if they choose to update the samplesresource such that creds are no longer needed, we would need to process/react to that and update status/proceed accordingly, which the code currently wouldn't do.

Reply:
Actually I spoke too soon ... the status update of the config map won't happen on the cached value, but on the event ... but I'll update the cache anyway, doesn't hurt, as well as update the comment, to provide some of the detail I noted here, as there is complexity here

Reply:
yeah since the sdk does not surface an UpdateStatus method, keeping the cache up to date with the most recent resource version is very helpful :-)